### ISSUE TITLE
Player's Unexpected physical behavior when located on a moving object.

### ISSUE THREAD

**Issue by luluxia** [replit/kaboom#763]
I would like to use moving platforms in the game. When a player jumps onto these platforms, the player can follow them along. As is shown in this [[demo](https://jsfiddle.net/y4kvecLg/2/)](https://jsfiddle.net/y4kvecLg/2/) I used before, when the `player` adds `z(1)`, it can follow the moving platforms along. This works on `kaboom@3000.0.0-beta.2`, but on versions newer than `kaboom@3000.0.0-beta.2`, something strange happens as the `onFallOff` and `onGround` get triggered repeatedly. I found the [[commit](https://github.com/replit/kaboom/commit/1739d72f5761b4783d1f6aaf6a33638f4de4c54c)](https://github.com/replit/kaboom/commit/1739d72f5761b4783d1f6aaf6a33638f4de4c54c) that causes this problem, but I have no idea how to fix it.

Also, I would like to ask whether the method in this demo is correct to make a movable platform?

**Comment by slmjkdbtl** [comment]
Yes that commit breaks moving platforms, just reverted in `v3000.1.13`

**Comment by luluxia** [comment]
thanks, It works now, but I've found that sometimes the z() component and the order in which objects are added still cause this problem when the player is standing on a vertically falling platform in my game. I tried to write the simplest case to reproduce the problem, but it hasn't worked yet.
